### ISSUE TITLE
Windows issue: Compilation as part of another program. Gflags stops it. 

### ISSUE THREAD

**Issue by nonspacial** [gflags/gflags#244]
Trying to Compile a prog called Carabiner that can be found here:

[https://github.com/brunchboy/carabiner](url)

I get this basically:

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

PS C:\WINDOWS\system32> cd C:\Repo\carabiner\build
PS C:\Repo\carabiner\build> cmake -G "MSYS Makefiles" ..
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: C:/MinGW/bin/gcc.exe
-- Check for working C compiler: C:/MinGW/bin/gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/MinGW/bin/g++.exe
-- Check for working CXX compiler: C:/MinGW/bin/g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include unistd.h
-- Looking for C++ include unistd.h - found
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
-- Looking for C++ include inttypes.h
-- Looking for C++ include inttypes.h - found
-- Looking for C++ include sys/types.h
-- Looking for C++ include sys/types.h - found
-- Looking for C++ include sys/stat.h
-- Looking for C++ include sys/stat.h - found
-- Looking for C++ include fnmatch.h
-- Looking for C++ include fnmatch.h - found
-- Looking for C++ include stddef.h
-- Looking for C++ include stddef.h - found
-- Check size of uint32_t
-- Check size of uint32_t - done
-- Looking for strtoll
-- Looking for strtoll - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Check size of pthread_rwlock_t
-- Check size of pthread_rwlock_t - failed
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Repo/carabiner/build
PS C:\Repo\carabiner\build> cmake --build .
Scanning dependencies of target gflags_nothreads_static
[  7%] Building CXX object gflags/CMakeFiles/gflags_nothreads_static.dir/src/gflags.cc.obj
In file included from C:/Repo/carabiner/build/gflags/include/gflags/config.h:115:0,
                 from c:/Repo/carabiner/gflags/src/gflags.cc:90:
C:/Repo/carabiner/gflags/src/windows_port.h: In function 'void setenv(const char*, const char*, int)':
C:/Repo/carabiner/gflags/src/windows_port.h:80:13: error: ambiguating new declaration of 'void setenv(const char*, const char*, int)'
 inline void setenv(const char* name, const char* value, int) {
             ^~~~~~
In file included from c:\mingw\lib\gcc\mingw32\6.3.0\include\c++\cstdlib:75:0,
                 from c:\mingw\lib\gcc\mingw32\6.3.0\include\c++\stdlib.h:36,
                 from C:/Repo/carabiner/gflags/src/windows_port.h:56,
                 from C:/Repo/carabiner/build/gflags/include/gflags/config.h:115,
                 from c:/Repo/carabiner/gflags/src/gflags.cc:90:
c:\mingw\include\stdlib.h:837:30: note: old declaration 'int setenv(const char*, const char*, int)'
 __cdecl __MINGW_NOTHROW  int setenv( const char *__n, const char *__v, int __f )
                              ^~~~~~
make[2]: *** [gflags/CMakeFiles/gflags_nothreads_static.dir/src/gflags.cc.obj] Error 1
make[1]: *** [gflags/CMakeFiles/gflags_nothreads_static.dir/all] Error 2
make: *** [all] Error 2
PS C:\Repo\carabiner\build>
```

It seems essentially the same prob as the issue below with the setenv() thing but it doesn't fix it if i copy paste what you said in that posts replies.

I'm using Powershell or Git Bash or Cmake GUI to the same effect. I have MinGW as the Compiler.

Thanks for your help in advance.

Robbie

**Comment by nonspacial** [comment]
If I copy and paste the files from the latest 2.2.1 stable release .zip I get 12% not 7% but it ignores pthreads in the files list which I presume is not helpful maybe? any way it looks like this for comparison with above:

```
PS C:\Repo\carabiner\build> cmake -G "MSYS Makefiles" ..
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: C:/MinGW/bin/gcc.exe
-- Check for working C compiler: C:/MinGW/bin/gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/MinGW/bin/g++.exe
-- Check for working CXX compiler: C:/MinGW/bin/g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include unistd.h
-- Looking for C++ include unistd.h - found
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
-- Looking for C++ include inttypes.h
-- Looking for C++ include inttypes.h - found
-- Looking for C++ include sys/types.h
-- Looking for C++ include sys/types.h - found
-- Looking for C++ include sys/stat.h
-- Looking for C++ include sys/stat.h - found
-- Looking for C++ include fnmatch.h
-- Looking for C++ include fnmatch.h - found
-- Looking for C++ include stddef.h
-- Looking for C++ include stddef.h - found
-- Check size of uint32_t
-- Check size of uint32_t - done
-- Looking for strtoll
-- Looking for strtoll - found
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Repo/carabiner/build
PS C:\Repo\carabiner\build> cmake --build .
Scanning dependencies of target gflags_nothreads_static
[ 12%] Building CXX object gflags/CMakeFiles/gflags_nothreads_static.dir/src/gflags.cc.obj
In file included from C:/Repo/carabiner/build/gflags/include/gflags/config.h:110:0,
                 from c:/Repo/carabiner/gflags/src/gflags.cc:90:
C:/Repo/carabiner/gflags/src/windows_port.h: In function 'void setenv(const char*, const char*, int)':
C:/Repo/carabiner/gflags/src/windows_port.h:80:13: error: ambiguating new declaration of 'void setenv(const char*, const char*, int)'
 inline void setenv(const char* name, const char* value, int) {
             ^~~~~~
In file included from c:\mingw\lib\gcc\mingw32\6.3.0\include\c++\cstdlib:75:0,
                 from c:\mingw\lib\gcc\mingw32\6.3.0\include\c++\stdlib.h:36,
                 from C:/Repo/carabiner/gflags/src/windows_port.h:56,
                 from C:/Repo/carabiner/build/gflags/include/gflags/config.h:110,
                 from c:/Repo/carabiner/gflags/src/gflags.cc:90:
c:\mingw\include\stdlib.h:837:30: note: old declaration 'int setenv(const char*, const char*, int)'
 __cdecl __MINGW_NOTHROW  int setenv( const char *__n, const char *__v, int __f )
                              ^~~~~~
make[2]: *** [gflags/CMakeFiles/gflags_nothreads_static.dir/src/gflags.cc.obj] Error 1
make[1]: *** [gflags/CMakeFiles/gflags_nothreads_static.dir/all] Error 2
make: *** [all] Error 2
```
Thanks again

**Comment by schuhschuh** [comment]
Can you post the relevant code snippets from `c:\mingw\lib\gcc\mingw32\6.3.0\include\c++\stdlib.h:36`?

I'd consider this a duplicate of #228, though.
